cuqi/solver/_solver.py

```diff
@@ -565,3 +565,102 @@ class PDHG(object):
     """Primal-Dual Hybrid Gradient algorithm."""
     def __init__(self):
         raise NotImplementedError
+
+
+
+class FISTA(object):
+    """Fast Iterative Shrinkage-Thresholding Algorithm for regularized least squares problems.
+
+    Minimize ||Ax-b||^2 + f(x).
+    
+    Parameters
+    ----------
+    A : ndarray or callable f(x,*args).
+    b : ndarray.
+    x0 : ndarray. Initial guess.
+    proximal : callable f(x, gamma) for proximal mapping.
+    maxit : The maximum number of iterations.
+    stepsize : The stepsize of the gradient step.
+    abstol : The numerical tolerance for convergence checks.
+    adapative : Whether to use FISTA or ISTA.
+    """  
```

Comment:
An example code block will be helpful here. For example (this is mainly similar to what is in the test)
```
    Example
    -----------
    .. code-block:: python

        From cuqi.solver import FISTA,  ProximalL1
        import scipy as sp

        m, n = 10, 5
        A = rng.standard_normal((m, n))
        b = rng.standard_normal(m)
        stepsize = 0.99/(sp.linalg.interpolative.estimate_spectral_norm(A)**2)
        x0 = np.zeros(n)
        fista = FISTA(A, b, x0, proximal = ProximalL1, stepsize = stepsize, maxit = 100, abstol=1e-12, adaptive = True)
        sol, _ = fista.solve()


```

Reply:
Done